iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java

```diff
@@ -493,20 +501,47 @@ private String getStatsSource() {
   }
 
   private Path getStatsPath(Table table) {
-    return new Path(table.location() + STATS + table.name() + table.currentSnapshot().snapshotId());
+    return getStatsPath(table, table.currentSnapshot().snapshotId());
+  }
+
+  private Path getStatsPath(Table table, long snapshotId) {
+    return new Path(table.location() + STATS + table.name() + snapshotId);
   }
 
-  private void invalidateStats(Path statsPath) {
+  private void checkAndMergeOrInvalidateStats(ColumnStatistics statsObjNew,
+      org.apache.hadoop.hive.ql.metadata.Table hmsTable) {
+    Table tbl = IcebergTableUtil.getTable(conf, hmsTable.getTTable());
```

Comment:
try to avoid loading table multiple times

Reply:
fixed